### PR TITLE
Add AWS Glue - Experimental

### PR DESCRIPTION
**Pull request by youenchene**

### CODE REVIEW

File: technologies/job/aws-glue/job/instanceActions.js [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+        return Response.success(JobStatus.QUEUED);
+      case 'FAILED':
+        return Response.success(JobStatus.FAILED);
+      case 'STOPPING':
+        return Response.success(JobStatus.KILLING);
+      case 'TIMEOUT':
+        return Response.success(JobStatus.FAILED);
+      default:
+        return Response.success(JobStatus.AWAITING);
+    }
```
Comment (yoannfleurydev) [inline technologies/job/aws-glue/job/instanceActions.js:83]:
```suggestion
	const JOB_STATES = {
		RUNNING: JobStatus.RUNNING,
		STOPPED: JobStatus.KILLED,
		SUCCEEDED: JobStatus.SUCCEEDED,
		STARTING: JobStatus.QUEUED,
		FAILED: JobStatus.FAILED,
		STOPPING: JobStatus.KILLING,
		TIMEOUT: JobStatus.FAILED,
	};
	
	return Response.success(JOB_STATES[data.JobRun.JobRunState] || JobStatus.AWAITING);
```

File: technologies/job/aws-glue/job/instanceActions.js [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    const cwl = new AWS.CloudWatchLogs({apiVersion: '2014-03-28'});
+
+    const params = {
+      logGroupName: '/aws-glue/jobs/output',
+      logStreamName: instance.payload.glueJobId
+    };
+
+    const logs = await cwl.getLogEvents(params).promise();
+
+    return Response.success(logs.events.map((item) => Log(item.message, 'stdout', new Date(item.timestamp*1000).toISOString())));
```
Comment (yoannfleurydev) [inline technologies/job/aws-glue/job/instanceActions.js:110]:
```suggestion
    return Response.success(logs.events.map((item) => Log(item.message, Stream.STDOUT, new Date(item.timestamp*1000).toISOString())));
```

File: technologies/job/aws-glue/job/instanceActions.js [hunk 3]
```diff
@@ -0,0 +1,115 @@
+const { Response, JobStatus, Log } = require('@saagie/sdk');
```
Comment (yoannfleurydev) [inline technologies/job/aws-glue/job/instanceActions.js:1]:
```suggestion
const { Response, JobStatus, Log, Stream } = require('@saagie/sdk');
```

File: technologies/job/aws-glue/workflow/instanceActions.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    const glue = new AWS.Glue({apiVersion: '2017-03-31'});
+
+    const data = await glue.getWorkflowRun({ Name: job.featuresValues.workflow.id, RunId: instance.payload.glueWorkflowId }).promise();
+
+    switch (data.Run.Status) {
+      case 'RUNNING':
+        return Response.success(JobStatus.RUNNING);
+      case 'COMPLETED':
+        return Response.success(JobStatus.SUCCEEDED);
+    }
```
Comment (yoannfleurydev) [inline technologies/job/aws-glue/workflow/instanceActions.js:48]: No default here ?

Comment (youenchene) [reply technologies/job/aws-glue/workflow/instanceActions.js:48]: No and AWAITING status put the user in error.

Comment (yoannfleurydev) [reply technologies/job/aws-glue/workflow/instanceActions.js:48]: OK, thanks :+1:

**Pull request merged**